http.ts

```diff
@@ -43,14 +43,19 @@ interface Response {
   body?: Uint8Array;
 }
 
-function setContentLength(r: Response): void {
+function setHeaders(r: Response): void {
   if (r.body) {
     if (!r.headers) {
       r.headers = new Headers();
     }
     if (!r.headers.has("content-length")) {
       r.headers.append("Content-Length", r.body.byteLength.toString());
     }
+    // set the content-type only if not yet set
+    if (!r.headers.has("content-type")) {
+      const body = new TextDecoder().decode(r.body);
+      r.headers.append("Content-Type", /^\s*</.test(body) ? 'html' : 'text');
```

Comment:
So this is looking to see if the body has a '<' leading character? That's not a valid way to check the content-type. Indeed, I don't think the server can know what content-type is being sent. That should be for the user to set.